generators/frontend/templates/react-demo/components/CategoriesList.js

```diff
@@ -0,0 +1,108 @@
+import React, { Component } from 'react';
+import CategoryCard from './CategoryCard';
+import Util from '../common/Util';
+import styled from 'styled-components';
+
+export default class extends Component {
+  state = {
+    categories: [],
+    productsList: [],
+    show: true
+  };
+
+  async componentDidMount() {
+    const cats = await Util.fetchCategories();
+    const prods = await Util.fetchProducts(cats[0].id);
+    this.setState(
+      {
+        categories: cats,
+        productsList: prods.results
+      }
+    );
+  }
+
+  onClickCategory = (e, categoryId) => {
+    e.preventDefault();
+    Util.fetchProducts(categoryId).then(list =>
+      this.setState({ productsList: list.results })
+    );
+  }
+
+  showProductsList = (e) => {
+    e.preventDefault();
+    this.setState({ show: false })
+  }
+
+  render() {
+    const { categories, productsList } = this.state;
+
+    const Wrapper = styled.div`
+      margin: 1.875rem auto;
+      padding: 1.25rem 2.5rem;
+      color: #2f4f4f;
```

Comment:
`#2f4f4f` is used in many places. It should better be set in a constant. Also, its repetition can probably be decreased by creating a generic css class.